templates/clusterclass-dev-test.yaml

```diff
@@ -0,0 +1,126 @@
+      - class: default-worker
+        template:
+          bootstrap:
+            ref:
+              apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+              kind: KubeadmConfigTemplate
+              name: dev-test-default-worker-bootstraptemplate
+          infrastructure:
+            ref:
+              apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+              kind: OpenStackMachineTemplate
+              name: dev-test-default-worker-machine-template
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
+metadata:
+  name: dev-test-default-worker-bootstraptemplate
+spec:
+  template:
+    spec:
+      files: []
+      joinConfiguration:
+        nodeRegistration:
+          kubeletExtraArgs:
+            cloud-provider: external
+            provider-id: openstack:///'{{ instance_id }}'
+          name: '{{ local_hostname }}'
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlaneTemplate
+metadata:
+  name: dev-test-control-plane-template
+spec:
+  template:
+    spec:
+      kubeadmConfigSpec:
+        clusterConfiguration:
+          apiServer:
+            extraArgs:
+              cloud-provider: external
+          controllerManager:
+            extraArgs:
+              cloud-provider: external
+        files: []
+        initConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: external
+              provider-id: openstack:///'{{ instance_id }}'
+            name: '{{ local_hostname }}'
+        joinConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: external
+              provider-id: openstack:///'{{ instance_id }}'
+            name: '{{ local_hostname }}'
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+kind: OpenStackClusterTemplate
+metadata:
+  name: dev-test-openstackcluster-template
+spec:
+  template:
+    spec:
+      apiServerLoadBalancer:
+        enabled: true
+      cloudName: capo-e2e
+      dnsNameservers:
+      - 8.8.8.8
+      identityRef:
+        kind: Secret
+        name: dev-test-clouds-yaml
+      managedSecurityGroups: true
+      nodeCidr: 10.6.0.0/24
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+kind: OpenStackMachineTemplate
+metadata:
+  name: dev-test-control-plane-machine-template
+spec:
+  template:
+    spec:
+      cloudName: capo-e2e
+      flavor: m1.medium
+      identityRef:
+        kind: Secret
+        name: dev-test-clouds-yaml
+      image: ubuntu-2204-kube-${KUBERNETES_VERSION}
+      sshKeyName: ${OPENSTACK_SSH_KEY_NAME}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+kind: OpenStackMachineTemplate
+metadata:
+  name: dev-test-default-worker-machine-template
+spec:
+  template:
+    spec:
+      cloudName: capo-e2e
+      flavor: m1.small
+      identityRef:
+        kind: Secret
+        name: dev-test-clouds-yaml
+      image: ubuntu-2204-kube-${KUBERNETES_VERSION}
```

Comment:
This version should be set using [ClusterClass variables](https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-class/write-clusterclass#builtin-variables). Otherwise it is hard coded once you render the ClusterClass, so all Clusters created from it will need to use the same version.